src/libcore/str/mod.rs

```diff
@@ -1416,11 +1455,12 @@ impl StrExt for str {
     }
 
     #[inline]
-    fn splitn<'a, P: Pattern<'a>>(&'a self, count: usize, pat: P) -> SplitN<'a, P> {
+    fn splitn<'a, P: Pattern<'a>>(&'a self, count: usize, pat: P) -> SplitN<'a, P>
+        where P::Searcher: Searcher<'a>
```

Comment:
Strictly, it is unnecessary, but having it here provide *much* better error messages. If you have this here, the error message lets you know that *this call* needs to be changed. Otherwise, the return type simply won't implement `Iterator` and you end up with "foo doesn't implement a method called {collect,map,etc}, try implementing Iterator".